runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -137,20 +137,7 @@ func TestCheckInvalidDictionaryKeys(t *testing.T) {
 
 	errs := ExpectCheckerErrors(t, err, 1)
 
-	assert.IsType(t, &sema.TypeMismatchError{}, errs[0])
-}
-
-func TestCheckInvalidDictionaryValues(t *testing.T) {
```

Comment:
Maybe don't remove this test case but turn it into a valid one, now that it is 🎉

Reply:
Added a set of tests under `TestCheckDictionarySupertypeInference` that covers all cases. So removed this since it's a duplicate now.